### DISCUSSION TITLE
Problem With Deploying MoodNft

### DISCUSSION THREAD

**Discussion by Play3rZer0** [Cyfrin/foundry-full-course-cu#2997] [General]
I am getting an error from my DeployMoodNft.s.sol script saying:

Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.

This referred to the particular line for:

function run() external returns(MoodNft)

I tried deploying the script and then it returns this error:

Warning (6321): Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
  --> script/DeployMoodNft.s.sol:11:37:
   |
11 |     function run() external returns(MoodNft) {
   |                                     ^^^^^^^

I cannot seem to find any similar problem to this in the discussion. 

Here is my complete code:
```

//SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Script, console} from "forge-std/Script.sol";
import {MoodNft} from "../src/MoodNft.sol";
import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";

contract DeployMoodNft is Script {

    function run() external returns(MoodNft) {
       string memory sadSvg = vm.readFile("./img/sad.svg");
       string memory happySvg = vm.readFile("./img/happy.svg");
       console.log(sadSvg);
       console.log(happySvg);
    }

    //automates base64 encoding
    function svgToImageURI(string memory svg) public pure returns(string memory) {
        string memory baseURL = "data:image/svg+xml;base64,";
        string memory svgBase64Encoded = Base64.encode(
            bytes(string(abi.encodePacked(svg)))
        );
        return string(abi.encodePacked(baseURL, svgBase64Encoded));
    }

}

```

**Comment by EngrPips** [comment]
Hello @Play3rZer0  That is just a warning and not an error. why did you declare that the `run` function will return a type of `MoodNft` but you did not return anything inside the `run` function implementation?

**Reply by Play3rZer0** [reply]
I am just following the code from the lesson. MoodNft was declared but the compiler does not like it and I cannot deploy anything.

**Reply by EngrPips** [reply]
Alright, Here is the GitHub repo https://github.com/Cyfrin/foundry-nft-cu that contains the codebase for the NFT section. Check it out and see where you are getting it wrong.

**Reply by Play3rZer0** [reply]
Ok, got it thanks. I will continue from there. I actually stopped at this part of the lesson and did not proceed realizing that it is not yet at the part where the contract is fully deployed. My bad on this one and case closed.

**Reply by EngrPips** [reply]
No worries, we live we learn.
